### ISSUE TITLE
Unclosed consumer after calling closeAsync

### ISSUE THREAD

**Issue by ntysdd** [apache/pulsar#3226]
#### Expected behavior

Consumer should be closed, and disappear in Pulsar Dashboard after we call closeAsync().

#### Actual behavior

Consumer still there, and if we create a new consumer to the topic (same subscription shared mode), we see two (and more and more) consumers (with same addresses and ports)

#### Steps to reproduce

Create two consumers (maybe in different JVMs) in shared mode to the same topic. One of them called unsubscriptionAsync(), when the future failed, call closeAsync(), it will succeed, but the consumer will remain in the Dashboard (and pulsar admin stats). Then create a new consumer in that process, will see two consumers.

#### System configuration
**Pulsar version**: 2.2.0
Java version: 1.8
OS: Linux

**Comment by xiangzilv123** [comment]
![71545303219_ pic](https://user-images.githubusercontent.com/1798683/50286586-d49c5280-049a-11e9-91f0-909b444f66ea.jpg)

**Comment by sijie** [comment]
@ntysdd @xiangzilv123 : the metrics are refreshed in a given interval. I don't remember it is 30 seconds or 5 mins. Can you wait for some time and verify if the metrics are refreshed?

**Comment by xiangzilv123** [comment]
we wait more than 30min
![101545357940_ pic_hd](https://user-images.githubusercontent.com/1798683/50320570-f97adf00-0507-11e9-9ee9-64e3877ef713.jpg)

**Comment by legendtkl** [comment]
hi, @xiangzilv123 , would you mind giving the code snippet ?

**Comment by xiangzilv123** [comment]
```java
package test;

import org.apache.pulsar.client.api.Consumer;
import org.apache.pulsar.client.api.PulsarClient;
import org.apache.pulsar.client.api.SubscriptionType;

import java.util.concurrent.TimeUnit;

public class TestMain {

    private static void waitTillNextMinute() throws InterruptedException {
        System.out.println("waiting for next minute");
        long m = System.currentTimeMillis() / (60 * 1000);
        do {
            Thread.sleep(1000);
        } while (System.currentTimeMillis() / (60 * 1000) == m);
    }

    public static void main(String[] args) throws Exception {
        String url = "pulsar://localhost:6650";
        String topic = "test-topic";
        String subscriptionName = "test-subscription";

        PulsarClient pulsarClient = PulsarClient.builder()
                .serviceUrl(url)
                .build();
        System.out.println("client created");

        while (true) {
            Consumer<byte[]> consumer = pulsarClient.newConsumer()
                    .topic(topic)
                    .ackTimeout(2 * 60, TimeUnit.MINUTES)
                    .subscriptionName(subscriptionName)
                    .subscriptionType(SubscriptionType.Shared)
                    .subscribe();
            System.out.println("consumer created");

            waitTillNextMinute();

            System.out.println("will unsub or close");
            Consumer<byte[]> ref = consumer;
            consumer.unsubscribeAsync().whenComplete((r, e) -> {
                if (e == null) {
                    System.out.println("unsub succeeded");
                } else {
                    e.printStackTrace();
                }
                ref.closeAsync().whenComplete((r2, e2) -> {
                    if (e2 == null) {
                        System.out.println("close succeeded");
                    } else {
                        e2.printStackTrace();
                    }
                });
            });

            waitTillNextMinute();
        }
    }
}

```

**Comment by xiangzilv123** [comment]
@legendtkl  you must start two consumers;    shared mode

**Comment by tuteng** [comment]
Reference https://github.com/apache/pulsar/pull/3236

**Comment by sijie** [comment]
@legendtkl is working on this issue.

**Comment by sijie** [comment]
added back to 2.3.0 since it is not difficult fix. 

@ntysdd : @legendtkl pinged me offline, he will pick this up again this week. so hopefully we will have the fix for 2.3.0.
